File: codility/MaxProductOfThree.java
package codility;

import java.util.Arrays;

public class MaxProductOfThree {

    public static void main(String[] args) {
        int[] arr = {-3,1,-2,2,5,6,4};
        System.out.println(maximumProduct(arr));
    }

    private static int maximumProduct(int[] arr) {
        Arrays.sort(arr);
        int maxproduct = 0;
        int n = arr.length;
        if((arr[0]*arr[1]*arr[n-1]) > (arr[n-1]*arr[n-2]*arr[n-3])){
            maxproduct = (arr[0]*arr[1]*arr[n-1]);
        }
        else{
            maxproduct = (arr[n-1]*arr[n-2]*arr[n-3]);
        }
        return maxproduct;
    }
}

File: codility/max_product_of_three.py
# -*- coding: utf-8 -*-
"""
Created on Fri Mar 16 23:48:32 2018

@author: daniels
"""

def max_product_of_three(arr):
    arr.sort()
    max_product = 0
    
    if((arr[-1]*arr[0]*arr[1])>(arr[-1]*arr[-2]*arr[-3])):
        return (arr[-1]*arr[0]*arr[1])
    else:
        return (arr[-1]*arr[-2]*arr[-3])
        
    return max_product